### DISCUSSION TITLE
Page numbering

### DISCUSSION THREAD

**Discussion by kyungnapark** [thegetty/quire#182] [Get Help] [answered]
I want the table of contents to list the page after it as the 2nd page, instead of the 8th page as it's doing right now. Any tips?

![Screen Shot 2021-09-19 at 9 23 06 AM](https://user-images.githubusercontent.com/87673056/133935082-781bb673-c980-4fdf-a9f6-6187091f8c39.png)

**Comment by geealbers** [comment]
Based on the screenshot here, that would mean the second page of the contents would be page 1? That would be very unconventional to start page numbering in the middle of the contents list. That said, if you want to do it, I think you could use the `counter-reset` CSS property. 

This is the CSS that's applied when you add `class: page-one` to a Markdown page in a Quire project:

```css
.quire-page--page-one {
    page: page-one;
    counter-reset: page 1;
    page-break-before: right;
  }
```

You could conceivably mark your contents page as `class: page-one` and then in custom.css add in this:

```css
.quire-page--page-one {
    counter-reset: page 0;
    page-break-before: left;
  }
```

At least I think that would do it. You'd need to test and maybe make some adjustments.

**Reply by kyungnapark** [reply]
Sorry, I've uploaded a more relevant screenshot.

So the goal is to have the table of contents list page 2 as what is labelled page 8 at the moment (the page after the 2nd page of the toc), essentially ignoring all pages from what is labelled page 2 to page 7 right now. 

Pages 3 and 5 are generated automatically and not accessible as markdown files, so I'm not sure how to get the table of contents to ignore those pages and skip right to the text on page 8.

![Screen Shot 2021-09-20 at 6 06 58 PM](https://user-images.githubusercontent.com/87673056/134097124-53db36fd-e75c-4388-bdc9-60c38f82730b.png)

**Comment by geealbers** [comment]
Does it work to tag the page after the contents, the page that is now page 8 and that you want to be page 2, with `class: page-one` in the YAML. And then in custom.css add:

```css
.quire-page--page-one {
   counter-reset: page 2;
   page-break-before: left;
 }
```

Or it might have to be `page-break-before: auto;`. Not sure. Other options at: https://developer.mozilla.org/en-US/docs/Web/CSS/page-break-before.

**Reply by kyungnapark** [reply]
Everything works except page 1 is listed as i, not 1. 
![Screen Shot 2021-09-21 at 4 50 55 PM](https://user-images.githubusercontent.com/87673056/134262315-37d1b34e-4ced-496e-b5a8-d87037d3ca5d.png)

**Comment by geealbers** [comment, marked as answer]
Since that's the cover, you wouldn't normally include it in a table of contents. So, I'd add `toc: false` to the markdown file for it.

(FWIW, the frontmatter of books are usually numbered in lowercase Roman numerals. The Arabic numeral 1 is then assigned to the first page of true "content" for the book. Usually the first chapter, or an Introduction. So, Quire handles things the same way, based on the placement of the `class: page-one` tag. For Quire, that tag means that this is page 1 (in Arabic numerals), and everything before it is frontmatter and so should be numbered in lowercase Roman.)

**Reply by kyungnapark** [reply]
I see! I'm aware listing the cover in the table of contents is unconventional--it's just that the client wanted the cover to be page 1 on the toc. Is it possible to make one last adjustment so that the i can be 1?

**Reply by geealbers** [reply]
Ok. The surest way would be to go into `themes/default/source/css/print.scss` and change every instance of `page, lower-roman` to just `page`. And on lines 194 and 208, you'll probably want to change `font-style: italic;` to `font-style: normal;`.

**Reply by kyungnapark** [reply]
Thank you!

**Comment by kyungnapark** [comment]
One last thing--I'm not sure why the page number on page 2 is not displaying, is this a custom.css issue?
![Screen Shot 2021-09-22 at 8 57 35 PM](https://user-images.githubusercontent.com/87673056/134452652-71218844-6b00-4039-a521-fc08ba5f8569.png)

**Reply by geealbers** [reply]
One of the consequences of overriding the default page numbering of `class: page-one`. But I think if you delete these lines from the `themes/default/source/css/print.scss` file (starting around line 144 I think) the number should display:

```scss
  @page page-one:first {
    @bottom-left {
      content: none;
    }
  }
```
